kubernetes/resource_kubernetes_role_test.go

```diff
@@ -0,0 +1,241 @@
+					resource.TestCheckResourceAttr("kubernetes_role.test", "rule.0.api_groups.#", "1"),
+					resource.TestCheckResourceAttr("kubernetes_role.test", "rule.0.api_groups.4161491668", "batch"),
+					resource.TestCheckResourceAttr("kubernetes_role.test", "rule.0.resources.#", "1"),
+					resource.TestCheckResourceAttr("kubernetes_role.test", "rule.0.resources.2828234181", "jobs"),
+					resource.TestCheckResourceAttr("kubernetes_role.test", "rule.0.verbs.#", "1"),
+					resource.TestCheckResourceAttr("kubernetes_role.test", "rule.0.verbs.1342917158", "watch"),
+					resource.TestCheckResourceAttr("kubernetes_role.test", "rule.0.resource_names.#", "0"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccKubernetesRole_importBasic(t *testing.T) {
+	resourceName := "kubernetes_role.test"
+	name := fmt.Sprintf("tf-acc-test-%s", acctest.RandStringFromCharSet(10, acctest.CharSetAlphaNum))
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckKubernetesRoleDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccKubernetesRoleConfig_basic(name),
+			},
+
+			{
+				ResourceName:            resourceName,
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"metadata.0.resource_version"},
+			},
+		},
+	})
+}
+
+func TestAccKubernetesRole_generatedName(t *testing.T) {
+	var conf api.Role
+	prefix := "tf-acc-test-gen-"
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:      func() { testAccPreCheck(t) },
+		IDRefreshName: "kubernetes_role.test",
+		Providers:     testAccProviders,
+		CheckDestroy:  testAccCheckKubernetesRoleDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccKubernetesRoleConfig_generatedName(prefix),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					testAccCheckKubernetesRoleExists("kubernetes_role.test", &conf),
+					resource.TestCheckResourceAttr("kubernetes_role.test", "metadata.0.annotations.%", "0"),
+					testAccCheckMetaAnnotations(&conf.ObjectMeta, map[string]string{}),
+					resource.TestCheckResourceAttr("kubernetes_role.test", "metadata.0.labels.%", "0"),
+					testAccCheckMetaLabels(&conf.ObjectMeta, map[string]string{}),
+					resource.TestCheckResourceAttr("kubernetes_role.test", "metadata.0.generate_name", prefix),
+					resource.TestMatchResourceAttr("kubernetes_role.test", "metadata.0.name", regexp.MustCompile("^"+prefix)),
+					resource.TestCheckResourceAttrSet("kubernetes_role.test", "metadata.0.generation"),
+					resource.TestCheckResourceAttrSet("kubernetes_role.test", "metadata.0.resource_version"),
+					resource.TestCheckResourceAttrSet("kubernetes_role.test", "metadata.0.self_link"),
+					resource.TestCheckResourceAttrSet("kubernetes_role.test", "metadata.0.uid"),
+				),
+			},
+		},
+	})
+}
+
+func testAccKubernetesRoleConfig_basic(name string) string {
+	return fmt.Sprintf(`
+resource "kubernetes_role" "test" {
```

Comment:
Would you please make sure all these HCL inline blocks are `terraform fmt` formatted?
Also, the last closing bracket of each inline resource should be followed by a newline.

This is necessary so that we can enable linting using this tool: https://github.com/katbyte/terrafmt 
You can also use that tool to do the formatting itself, as long as the "newline" rule is in place already.

Here's a cleanup PR for reference: https://github.com/terraform-providers/terraform-provider-kubernetes/pull/230